### PR TITLE
Helps 778 add rich text editor component

### PR DESCRIPTION
**Pull request by MaggieMaSS**
This is the test of RichTextEditor component

It includes a RichTextEditor. 
RichTextEditor Component

The RichTextEditor Component includes following features in toolbar:
 
- Bold

- Italics

- Numbered Lists

- Bulleted Lists

- Hyperlinks

- Strikethrough

- Indenting options

RichTextEditor includes onChange, onFocus and onBlur

Text will be translated to markdown after clicking save button

RichTextEditor provides customized validation error, loop authors can provide warning if text length hits the limit

Tooltip for the RichTextEditor

RichTextEditor supports flex layout(but since there's only one component in on the whisper so we don't add flex attribute here)

### CODE REVIEW

File: ldk/javascript/examples/self-test-loop/src/tests/whisper/index.ts [hunk 1]
```diff
@@ -1667,3 +1668,66 @@ export const testFlex = (): Promise<boolean> =>
...
+          type: WhisperComponentType.RichTextEditor,
+          onBlur: () => {
+            console.debug(`On blur called`);
+          },
+          onChange: (_error: Error, value: string) => {
+            console.debug(`Input value changed: ${value}`);
+            editedText = value;
+          },
+          onFocus: () => {
+            console.debug(`On blur called`);
```
Comment (RusGraf) [inline ldk/javascript/examples/self-test-loop/src/tests/whisper/index.ts:1689]: perhaps `On focus called`.

File: ldk/javascript/examples/self-test-loop/src/tests/whisper/index.ts [hunk 2]
```diff
@@ -1667,3 +1668,66 @@ export const testFlex = (): Promise<boolean> =>
...
+              label: 'Save',
+              onClick: (_error: Error, onClickWhisper: Whisper) => {
+                if (!editedText || editedText.length === 0 || editedText.length > 50) {
+                  (components[0] as RichTextEditor).validationError =
+                    'Inputed text is required and should be less than 200 chars.';
+                  onClickWhisper.update({
+                    components,
+                  });
+                } else {
+                  (components[0] as RichTextEditor).validationError = null;
```
Comment (RusGraf) [inline ldk/javascript/examples/self-test-loop/src/tests/whisper/index.ts:1709]: This line is not needed as we are not using components bellow. It is probably my oversight, please remove it ;)

File: ldk/javascript/src/types/oliveHelps/index.d.ts [hunk 3]
```diff
@@ -316,6 +316,8 @@ declare namespace OliveHelps {
     | 'radioGroup'
     | 'select'
     | 'sectionTitle'
+    | 'dateTimeInput'
```
Comment (RusGraf) [inline ldk/javascript/src/types/oliveHelps/index.d.ts:319]: Looks like you duplicating 'dateTimeInput' here. Please delete.

File: ldk/javascript/src/whisper/types.ts [hunk 4]
```diff
@@ -313,6 +324,8 @@ export type ChildComponents =
   | RadioGroup
   | Select
   | SectionTitle
+  | DateTimeInput
```
Comment (RusGraf) [inline ldk/javascript/src/whisper/types.ts:327]: Same in here you dont need to duplicate DateTimeInput its already presented on the line 314.

Comment (MaggieMaSS) [reply ldk/javascript/src/whisper/types.ts]: @RusGraf I have updated the code and get your comments solved :)

**Pull request merged**